File: jspweb/src/main/webapp/js/sample/sample.js

//1.

$.ajax({
	url : "/jspweb/Sample",
	data:{ type: 1 },
	method:"get",
	success : r => {
		console.log('성공');
	}
})

//2.
$.ajax({
	url : "/jspweb/Sample",
	data:{ type: 2 , slat , slng , elat , elng },
	method:"get",
	success : r => {
		console.log('성공');
	}
})

//3.
$.ajax({
	url : "/jspweb/Sample",
	data: { type: 3 , pno : pno },
	method:"get",
	success : r => {
		console.log('성공');
	}
})

//4.
$.ajax({
	url : "/jspweb/Sample",
	data: {type: 4 , page : 1 , listsize : 5 },
	method:"get",
	success : r => {
		console.log('성공');
	}
})

File: jspweb/src/main/java/controller/sample/Sample.java
package controller.sample;

import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import com.fasterxml.jackson.databind.ObjectMapper;


@WebServlet("/Sample")
public class Sample extends HttpServlet {
	private static final long serialVersionUID = 1L;
       

    public Sample() {
        super();
    }


	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		String type = request.getParameter("type");
		ObjectMapper objectMapper = new ObjectMapper();
		String json = "";
		
		if ( type.equals("1")) { // 타입 1 이면

			
		} else if ( type.equals("2")) { // 타입 2 이면
			String slat = request.getParameter("slat");
			String slng = request.getParameter("slng");
			String elat = request.getParameter("elat");
			String elng = request.getParameter("elng");
		} else if ( type.equals("3")) { // 타입 3 이면
			int pno = Integer.parseInt(request.getParameter("pno"));
			
		} else if ( type.equals("4")) { // 타입 4 이면
			int page = Integer.parseInt(request.getParameter("page"));
			int listsize = Integer.parseInt(request.getParameter("listsize"));
		}
		
	}


	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {

	}


	protected void doPut(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {

	}


	protected void doDelete(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {

	}

}
